### ISSUE TITLE
Persisent Storage for plugins

### ISSUE THREAD

**Issue by odilitime** [elizaOS/eliza#617]
**Is your feature request related to a problem? Please describe.**

Plugin developers needs a framework for storing persistent data for their plugins. Some data needs to be tied to an agent, other data to the nodejs instance. 

**Describe the solution you'd like**

I'd like to hear how plugin authors are currently tackling this and put together a workgroup to develop a long term plan.

**Describe alternatives you've considered**

Some options:

1. Decentralized: They could each have their own store
2. Centralized: We develop a universal key/value store OR registry

**Additional context**

None at this time

**Comment by lalalune** [comment]
Yeah, this is the same for caching and other stuff. We can add this to the MemoryManager, or make a DataManager that also uses the same adapters.
